pkg/auth/oauth/handlers/selectprovider.go

```diff
@@ -0,0 +1,186 @@
+package handlers
+
+// TODO JWF dont like that this is currently in handlers but it needs access to stuff in handlers and stuff in handlers needs access to it so would have to move some types into a separate package to move it out
```

Comment:
@liggitt select providers needed access to the AuthencationRedirector type in the handlers package (although that may change dependent on other questions forthcoming) but then the default_auth_handler stuff is also in the handlers package, so i couldn't move select provider stuff out without creating an import cycle.  Assuming if we wanted it in its own package we'd have to move the type defs to a different package?  How set are you on having it in its own package.